tests/network/probes.go

```diff
@@ -227,6 +229,27 @@ var _ = SIGDescribe("[ref_id:1182]Probes", func() {
 			Entry("[test_id:TODO]with working Exec probe", createExecProbe(period, initialSeconds, timeoutSeconds, "uname", "-a"), blankIPFamily),
 		)
 
+		Context("guest agent ping", func() {
+			BeforeEach(func() {
+				vmi = libvmi.NewFedora(withMasqueradeNetworkingAndFurtherUserConfig(
+					withLivelinessProbe(createGuestAgentPingProbe(period, initialSeconds)),
+				)...)
+				vmi = tests.VMILauncherIgnoreWarnings(virtClient)(vmi)
+
+				By("Waiting for agent to connect")
+				tests.WaitAgentConnected(virtClient, vmi)
+				Expect(console.LoginToFedora(vmi)).To(Succeed())
+			})
+
+			It("[test_id:TODO] VM stops when guest agent is disabled", func() {
+				Expect(stopGuestAgent(vmi)).To(Succeed())
+
+				Eventually(func() (*v1.VirtualMachineInstance, error) {
+					return virtClient.VirtualMachineInstance(util.NamespaceTestDefault).Get(vmi.Name, &metav1.GetOptions{})
+				}, 2*time.Minute, 1*time.Second).Should(Or(matcher.BeInPhase(v1.Failed), matcher.HaveSucceeded()))
```

Comment:
Huh, unrelated as this is my first time really looking at the behaviour of Liveness probes but does it really leave the VMI in a `v1.Failed` condition?

Reply:
Answering own question yes it does, apologies the noise!